File: src/main/java/de/esymetric/jerusalem/osmDataRepresentation/OSMWay.java
package de.esymetric.jerusalem.osmDataRepresentation;

import java.util.ArrayList;
import java.util.Map;

import de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.MemoryArrayOsmNodeID2OwnIDMap;

public class OSMWay {
	public int id;
	public ArrayList<Long> nodes;  // osm ids of nodes
	public Map<String, String> tags;
	public int wayCostIDForward, wayCostIDBackward;

	private short latLonDirKey;

	public short getLatLonDirID(MemoryArrayOsmNodeID2OwnIDMap osmID2ownIDMap) {  // int to short 25.03.13
		if (latLonDirKey == 0 && nodes.size() > 0 ) {
			latLonDirKey = osmID2ownIDMap.getShortCellID(nodes.get(0));
		}
		return latLonDirKey;
	}
}

File: src/main/java/de/esymetric/jerusalem/osmDataRepresentation/OSMWay.kt
package de.esymetric.jerusalem.osmDataRepresentation

import de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.MemoryArrayOsmNodeID2OwnIDMap

class OSMWay {
    var id = 0
    var nodes // osm ids of nodes
            : ArrayList<Long?>? = null
    var tags: MutableMap<String?, String?>? = null
    var wayCostIDForward = 0
    var wayCostIDBackward = 0
    private var latLonDirKey: Short = 0
    fun getLatLonDirID(osmID2ownIDMap: MemoryArrayOsmNodeID2OwnIDMap): Short {  // int to short 25.03.13
        if (latLonDirKey.toInt() == 0 && nodes!!.size > 0) {
            latLonDirKey = osmID2ownIDMap.getShortCellID(nodes!![0]!!)
        }
        return latLonDirKey
    }
}